mmpose/models/keypoint_heads/heatmap_3d_head.py

```diff
@@ -0,0 +1,222 @@
+        num_deconv_kernels (list|tuple): Kernel sizes.
+        in_index (int|Sequence[int]): Input feature index. Default: -1
+        input_transform (str|None): Transformation type of input features.
+            Options: 'resize_concat', 'multiple_select', None.
+            'resize_concat': Multiple feature maps will be resize to the
+                same size as first one and than concat together.
+                Usually used in FCN head of HRNet.
+            'multiple_select': Multiple feature maps will be bundle into
+                a list and passed into decode head.
+            None: Only one select feature map is allowed.
+            Default: None.
+        align_corners (bool): align_corners argument of F.interpolate.
+            Default: False.
+        loss_keypoint (dict): Config for keypoint loss. Default: None.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 depth_size=64,
+                 num_deconv_layers=3,
+                 num_deconv_filters=(256, 256, 256),
+                 num_deconv_kernels=(4, 4, 4),
+                 extra=None,
+                 in_index=0,
+                 input_transform=None,
+                 align_corners=False,
+                 loss_keypoint=None,
+                 train_cfg=None,
+                 test_cfg=None):
+
+        super().__init__(in_channels, out_channels, num_deconv_layers,
+                         num_deconv_filters, num_deconv_kernels, extra,
+                         in_index, input_transform, align_corners,
+                         loss_keypoint, train_cfg, test_cfg)
+        assert out_channels % depth_size == 0
+        self.depth_size = depth_size
+
+    def get_loss(self, output, target, target_weight):
+        """Calculate 3D heatmap loss.
+
+        Note:
+            batch_size: N
+            num_keypoints: K
+            heatmaps depth size: D
+            heatmaps height: H
+            heatmaps weight: W
+
+        Args:
+            output (torch.Tensor[NxKxDxHxW]): Output heatmaps.
+            target (torch.Tensor[NxKxDxHxW]): Target heatmaps.
+            target_weight (torch.Tensor[NxKx1]):
+                Weights across different joint types.
+        """
+        losses = dict()
+        assert not isinstance(self.loss, nn.Sequential)
+        assert target.dim() == 5 and target_weight.dim() == 3
+        losses['heatmap_loss'] = self.loss(output, target, target_weight)
+        return losses
+
+    # The accuracy of interhand keypoints is not well defined.
+    def get_accuracy(self, output, target, target_weight):
+        return {}
+
+    def forward(self, x):
+        """Forward function."""
+        x = super().forward(x)
+        N, C, H, W = x.shape
+        # reshape the 2D heatmap to 3D heatmap
+        x = x.reshape(N, C // self.depth_size, self.depth_size, H, W)
+        return x
+
+    def decode(self, img_metas, output, **kwargs):
```

Comment:
Will it error if `kwargs` is removed?

Reply:
Some heads need extra information for decode function, for example img_size. So I keep _kwargs_ to make the decode function in the same format.

Reply:
Okay. Lets keep it for now.